storefront-reference-architecture-master/cartridges/int_external_customers_data/cartridge/controllers/Account.js

```diff
@@ -0,0 +1,404 @@
+            registrationForm.login.password.valid = false;
+            registrationForm.login.passwordconfirm.valid = false;
+            registrationForm.login.passwordconfirm.error = Resource.msg(
+                "error.message.mismatch.password",
+                "forms",
+                null
+            );
+            registrationForm.valid = false;
+        }
+
+        if (
+            !CustomerMgr.isAcceptablePassword(
+                registrationForm.login.password.value
+            )
+        ) {
+            registrationForm.login.password.valid = false;
+            registrationForm.login.passwordconfirm.valid = false;
+            registrationForm.login.passwordconfirm.error = Resource.msg(
+                "error.message.password.constraints.not.matched",
+                "forms",
+                null
+            );
+            registrationForm.valid = false;
+        }
+
+        // setting variables for the BeforeComplete function
+        var registrationFormObj = {
+            firstName: registrationForm.customer.firstname.value,
+            lastName: registrationForm.customer.lastname.value,
+            phone: registrationForm.customer.phone.value,
+            email: registrationForm.customer.email.value,
+            emailConfirm: registrationForm.customer.emailconfirm.value,
+            password: registrationForm.login.password.value,
+            passwordConfirm: registrationForm.login.passwordconfirm.value,
+            validForm: registrationForm.valid,
+            form: registrationForm,
+        };
+
+        if (registrationForm.valid) {
+            res.setViewData(registrationFormObj);
+
+            this.on("route:BeforeComplete", function (req, res) {
+                // eslint-disable-line no-shadow
+                var Transaction = require("dw/system/Transaction");
+                var accountHelpers = require("*/cartridge/scripts/helpers/accountHelpers");
+                var authenticatedCustomer;
+                var serverError;
+
+                // getting variables for the BeforeComplete function
+                var registrationForm = res.getViewData(); // eslint-disable-line
+
+                if (registrationForm.validForm) {
+                    var login = registrationForm.email;
+                    var password = registrationForm.password;
+                    var serviceHook = HookMgr.hasHook(
+                        "app.external.customers.data"
+                    );
+                    const uniqueExternalId = UUIDUtils.createUUID();
+                    var response;
+
+                    if (serviceHook) {
+                        response = HookMgr.callHook(
+                            "app.external.customers.data",
+                            "createUser",
+                            uniqueExternalId,
+                            registrationForm
+                        );
+                    }
+
+                    if (!response.isOk() && serviceHook) {
```

Comment:
Should change the place within the condition as if serviceHook is false than response doesn't have isOk()